### ISSUE TITLE
Circular dependency in pysmt.typing?

### ISSUE THREAD

**Issue by rene-rex** [pysmt/pysmt#405]
Hello

I get an error on import of the typing module:

```python3 -c "import pysmt.typing"```
```python
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/rex/software/venvs/opt/lib/python3.5/site-packages/pysmt/typing.py", line 500, in <module>
    import pysmt.environment
  File "/home/rex/software/venvs/opt/lib/python3.5/site-packages/pysmt/environment.py", line 24, in <module>
    import pysmt.simplifier
  File "/home/rex/software/venvs/opt/lib/python3.5/site-packages/pysmt/simplifier.py", line 22, in <module>
    import pysmt.typing as types
AttributeError: module 'pysmt' has no attribute 'typing'
```
Maybe this is circular dependency? `types` imports `environment`, which imports `simplifier`, which in turn import `types` again.

**Comment by mikand** [comment]
Hi,

thanks for the report. Indeed this is a circular dependency that needs to be resolved.

We never realized this issue because it suffices to import `pysmt.environment` or `pysmt.shortcuts` before importing  `pysmt.types` for the problem to go away.

**Comment by marcogario** [comment]
Fixed in #406.
